system/src/system_sleep.cpp

```diff
@@ -81,13 +93,20 @@ int system_sleep_ext(const hal_sleep_config_t* config, hal_wakeup_source_base_t*
         spark_cloud_flag_disconnect();
     }
 
+    // TODO: restore network state if network is disconnected but it failed to enter sleep mode.
```

Comment:
We are still restoring it now, right? `system_sleep_network_suspend()` still returns `resume = true` even if something goes wrong.

Reply:
Just a reminder and in case that we use `CHECK()` in this function.